frame/system/src/limits.rs

```diff
@@ -407,20 +418,34 @@ impl BlockWeightsBuilder {
 
 		// compute max block size.
 		for class in DispatchClass::all() {
-			weights.max_block = match weights.per_class.get(*class).max_total {
-				Some(max) => max.max(weights.max_block),
-				_ => weights.max_block,
```

Comment:
The logic here was a bit weird. `None` for the class `max_total` was defined as *unlimited*, but it is not added as `u64:MAX` here to the `max_block` but just ignored instead.  
If I add it as `u64::MAX`, then every block is immediately `(u64::MAX, u64::MAX)` since we set the default for class `Mandatory` to `UNLIMITED` (same as legacy behaviour).  
Now this leads to the case that Polkadot ends up with `max_block` = `(2 seconds, 0 bytes)`, which is weird since its proof size is unlimited. Currently we have `(2 seconds, u64::MAX)`. This in itself is not an issue for the functionality of this MR, but it looks off. 

Not sure how to best solve this. Polkadot currently has `Mandatory = (u64::MAX, u64::MAX)` so the `max_block` *must* be `(u64::MAX, u64::MAX)`…

Reply:
After thinking again: Its probably enough to add a new `block_target` weight, which could then be set to `(2  seconds, 5 MiB)` on Polkadot.  
This is better than hard-coding it in frame-system and the tx-payment pallet. Changing `max_block` to be the *real* max block aka `max_value()` in most cases would be a big breaking change though. Or I do some cheats like using the soft limit if it would bw infinity otherwise…